machine/v1beta1/types_machineset.go

```diff
@@ -150,6 +150,11 @@ type MachineSetStatus struct {
 	// +optional
 	ErrorMessage *string `json:"errorMessage,omitempty"`
 
+	// Conditions defines the current state of the MachineSet
+	// +listType=map
+	// +listMapKey=type
+	Conditions []Condition `json:"conditions,omitempty"`
```

Comment:
Can we use metav1.Condition instead? What's the difference between our Condition type and the metav1 type?

Reply:
I'll have a look - I just used the same condition type as we use for `Machine` to keep things consistent :)

Reply:
https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L1532-L1578C1  vs https://github.com/openshift/api/blob/master/machine/v1beta1/types_provider.go#L195-L229 

In `metav1.Condition`, we have an `ObservedGeneration` that we do not in ours. In ours we have a Severity